modAionImpl/src/org/aion/zero/impl/sync/FastSyncManager.java

```diff
@@ -14,15 +37,56 @@
  */
 public final class FastSyncManager {
 
+    // TODO: ensure correct behavior when disabled
     private boolean enabled;
+    // TODO: ensure correct behavior when complete
     private final AtomicBoolean complete = new AtomicBoolean(false);
+    private final AtomicBoolean completeBlocks = new AtomicBoolean(false);
+
+    private final AionBlockchainImpl chain;
+    private final BlockHeaderValidator<A0BlockHeader> blockHeaderValidator;
+    private final P2pMgr p2pMgr;
+
+    // TODO: consider adding a FAST_SYNC log as well
+    private static final Logger log = AionLoggerFactory.getLogger(LogEnum.SYNC.name());
+
+    private AionBlock pivot = null;
+
+    Map<ByteArrayWrapper, Long> importedBlockHashes =
+            Collections.synchronizedMap(new LRUMap<>(4096));
+    Map<ByteArrayWrapper, ByteArrayWrapper> receivedBlockHashes =
+            Collections.synchronizedMap(new LRUMap<>(1000));
+
+    BlockingQueue<BlocksWrapper> downloadedBlocks = new LinkedBlockingQueue<>();
+    Map<ByteArrayWrapper, BlocksWrapper> receivedBlocks = new HashMap<>();
 
     private final Map<ByteArrayWrapper, byte[]> importedTrieNodes = new ConcurrentHashMap<>();
 
-    public FastSyncManager() {
-        this.enabled = false;
+    public FastSyncManager(
```

Comment:
Can we make it so that thread pool size and/or thread priority can be specified?

Seems a bit weird for a class to set that for itself.

Reply:
Like I mentioned in the PR description this class is far from finalized. It will have a dedicated PR when finalized.